satpy/readers/mws_l1b.py

```diff
@@ -0,0 +1,288 @@
+        return variable
+
+    def _get_dataset_channel(self, key, dataset_info):
+        """Load dataset corresponding to channel measurement.
+
+        Load a dataset when the key refers to a measurand, whether uncalibrated
+        (counts) or calibrated in terms of brightness temperature or radiance.
+
+        """
+        # Get the dataset
+        # Get metadata for given dataset
+        grp_pth = dataset_info['file_key']
+        channel_index = get_channel_index_from_name(key['name'])
+
+        data = self[grp_pth][:, :, channel_index]
+        attrs = data.attrs.copy()
+
+        fv = attrs.pop(
+            "FillValue",
+            default_fillvals.get(data.dtype.str[1:], np.nan))
+        vr = attrs.get("valid_range", [-np.inf, np.inf])
+
+        if key['calibration'] == "counts":
+            attrs["_FillValue"] = fv
+            nfv = fv
+        else:
+            nfv = np.nan
+        data = data.where(data >= vr[0], nfv)
+        data = data.where(data <= vr[1], nfv)
+
+        # Manage the attributes of the dataset
+        data.attrs.setdefault('units', None)
+        data.attrs.update(dataset_info)
+
+        dataset_attrs = getattr(data, 'attrs', {})
+        dataset_attrs.update(dataset_info)
+        dataset_attrs.update({
+            "platform_name": self.platform_name,
+            "sensor": self.sensor,
+            "orbital_parameters": {'sub_satellite_latitude_start': self.sub_satellite_latitude_start,
+                                   'sub_satellite_longitude_start': self.sub_satellite_longitude_start,
+                                   'sub_satellite_latitude_end': self.sub_satellite_latitude_end,
+                                   'sub_satellite_longitude_end': self.sub_satellite_longitude_end},
+        })
+
+        try:
+            dataset_attrs.update(key.to_dict())
+        except AttributeError:
+            dataset_attrs.update(key)
+
+        data.attrs.update(dataset_attrs)
+        return data
+
+    def _get_dataset_aux_data(self, dsname):
+        """Get the auxiliary data arrays using the index map."""
+        # Geolocation and navigation data:
+        if dsname in ['mws_lat', 'mws_lon',
+                      'solar_azimuth', 'solar_zenith',
+                      'satellite_azimuth', 'satellite_zenith',
+                      'surface_type', 'terrain_elevation']:
+            var_key = AUX_DATA.get(dsname)
+        else:
+            raise NotImplementedError(f"Dataset '{dsname}' not supported!")
+
+        try:
+            variable = self[var_key]
+        except KeyError:
+            logger.exception("Could not find key %s in NetCDF file, no valid Dataset created", var_key)
+            raise
+
+        # Scale the data:
+        if 'scale_factor' in variable.attrs and 'add_offset' in variable.attrs:
+            missing_value = variable.attrs['missing_value']
+            variable.data = da.where(variable.data == missing_value, np.nan,
```

Comment:
Is this tested?

Reply:
Probably not. I think this code should never be used as all datasets will have a scale and ofsset in the netCDF file

Reply:
Forget what I said above! The code is now tested, thx!